backend/compact-connect/lambdas/provider-data-v1/handlers/ingest.py

```diff
@@ -0,0 +1,128 @@
+from typing import Iterable
+
+from boto3.dynamodb.types import TypeSerializer
+
+from config import logger, config
+from data_model.schema.license import LicensePostSchema, LicenseRecordSchema
+from data_model.schema.provider import ProviderRecordSchema
+from exceptions import CCNotFoundException
+from handlers.utils import sqs_handler
+
+
+license_schema = LicensePostSchema()
+
+
+@sqs_handler
+def ingest_license_message(message: dict):
+    """
+    For each message, validate the license data and persist it in the database
+    """
+    # This should already have been validated at this point, before the data was ever sent for ingest,
+    # but validation is cheap. We can do it again, just to protect ourselves from something unexpected
+    # happening on the way here.
+    license_post = license_schema.load(message['detail'])
+    compact = license_post['compact']
+    jurisdiction = license_post['jurisdiction']
+
+    provider_id = config.data_client.get_or_create_provider_id(compact=compact, ssn=license_post['ssn'])
+    logger.info('Updating license data', provider_id=provider_id, compact=compact, jurisdiction=jurisdiction)
+
+    # Start preparing our db transactions
+    dynamo_transactions = [
+        # Put the posted license
+        {
+            'Put': {
+                'TableName': config.provider_table_name,
+                # We'll use the schema/serializer to populate index fields for us
+                'Item': TypeSerializer().serialize(LicenseRecordSchema().dump({
+                    'providerId': provider_id,
+                    'compact': compact,
+                    'jurisdiction': jurisdiction,
+                    **license_post
+                }))['M']
+            }
+        }
+    ]
+
+    try:
+        provider_data = config.data_client.get_provider(  # pylint: disable=missing-kwoa,unexpected-keyword-arg
+            compact=compact,
+            provider_id=provider_id,
+            detail=True,
+            consistent_read=True
+        )
+        # Get all privilege jurisdictions, directly from privilege records
+        privilege_jurisdictions = {
+            record['jurisdiction']
+            for record in provider_data['items']
+            if record['type'] == 'privilege' and record['status'] == 'active'
+        }
+        # Get all the existing license records, by jurisdiction, to find the best data for the provider
+        licenses = {
+            record['jurisdiction']: record
+            for record in provider_data['items']
+            if record['type'] == 'license'
+        }
```

Comment:
There cannot be multiple licenses for the same provider (by SSN) from a given jurisdiction - it's built into the the primary id for the data model. If a state attempts to write multiple licenses for the same SSN, they'll be overwriting one record with another.

Reply:
Hmm, I recall a test that added an "inactive" license. What happens if I have a license in OH, then - for whatever reason - the compact uploads an _inactive_ license for me in OH. My understanding is that this inactive license still gets added to my list of licenses, but now I have two licenses in OH. Or, since it's the same jurisdiction, will the inactive license overwrite the active one?

Reply:
There is a test that adds an 'inactive' license, but from a second jurisdiction. If you have an active license in OH and OH uploads again with an 'inactive' state, they are updating your license to deactivate it.

Reply:
Got it. Helpful. Thank you.